src/content/en/fundamentals/native-hardware/capturing-images/index.md

```diff
@@ -2,210 +2,343 @@ project_path: /web/_project.yaml
+Since most of the above methods produce a `FileList`, we should talk a little about what that is.
+
+A `FileList` is similar to an `Array`. It has numeric keys and a `length` property, but it isn't
+*actually* an array. There are no array methods, like `forEach` or `pop`, and it isn't iterable.
+Of course, you can get a real Array by using `Array.from(fileList)`.
+
+The entries of the `FileList` are `File` objects. These are exactly the same as `Blob` objects
+except that they have additional `name` and `lastModified` read-only properties.
+
+    <img id="output">
+    <script>
+      const output = document.getElementById('output');
+
+      function doSomethingWithFiles(fileList) {
+        let file = null;
+
+        for (let i = 0; i < fileList.length; i++) {
+          if (fileList[i].type.match(/^image\//)) {
+            file = fileList[i];
+            break;
+          }
+        }
+
+        if (file !== null) {
+          output.src = URL.createObjectURL(file);
+        }
+      }
+    </script>
+
+This example finds the first file that has an image MIME type, but you could also handle multiple
+images being selected/pasted/dropped at once.
+
+Once you have access to the file you can do anything you want with it. For
+example, you can:
+
+- Draw it into a `<canvas>` element so that you can manipulate it
+- Download it to the user's device
+- Upload it to a server with `fetch`
 
 ## Access the camera interactively
+Now that we've covered our bases, it's time to progressively enhance!
 
 Modern browsers can get direct access to cameras, allowing us to build
 experiences that are fully integrated with the web page, so the user need never
 leave the browser.
 
-Warning: Direct access to the camera is a powerful feature, it requires consent 
-from the user and your site needs to be on a secure origin (HTTPS).
-
 ### Acquire access to the camera
 
-We can directly access a camera and microphone by using an API in the WebRTC 
-specification called `getUserMedia()`. This will prompt the user for 
+We can directly access a camera and microphone by using an API in the WebRTC
+specification called `getUserMedia()`. This will prompt the user for
 access to their connected microphones and cameras.
 
+Support for `getUserMedia` is pretty good, but it isn't yet everywhere. In particular, it is not
+available in Safari 10 or lower, which at the time of writing is still the latest stable version.
+However, [Apple have announced](https://webkit.org/blog/7726/announcing-webrtc-and-media-capture/)
+that it will be available in Safari 11.
+
+It's very simple to detect support, however.
+
+    const supported = 'mediaDevices' in navigator;
+
+Warning: Direct access to the camera is a powerful feature. It requires consent
+from the user, and your site MUST be on a secure origin (HTTPS).
+
+When you call `getUserMedia`, you need to pass in an object that describes what kind of media you
+want. These choices are called constraints.
+
```

Comment:
Combine this and the next into one paragraph. It'll make the standalone at line 207 a little stronger.